File: src/main/scala/arimitsu/sf/cql/v3/message/Options.scala
package arimitsu.sf.cql.v3.message

;


import arimitsu.sf.cql.v3.{Header, Frame, Opcode, Version}
;

/**
 * Created by sxend on 14/06/07.
 */
case class Options(streamId: Short, flags: Byte) extends Request {
  def toFrame: Frame = {
    Frame(Header(Version.REQUEST, flags, streamId, Opcode.OPTIONS), None)
  }
}

File: src/main/scala/arimitsu/sf/cql/v3/messages/Options.java
package arimitsu.sf.cql.v3.messages;


import arimitsu.sf.cql.v3.Flags;
import arimitsu.sf.cql.v3.Frame;
import arimitsu.sf.cql.v3.Header;
import arimitsu.sf.cql.v3.Opcode;
import arimitsu.sf.cql.v3.Version;

/**
 * Created by sxend on 14/06/07.
 */
public class Options implements Request {
    public final short streamId;
    public final Flags flags;

    public Options(short streamId, Flags flags) {
        this.streamId = streamId;
        this.flags = flags;
    }

    @Override
    public Frame toFrame() {
        return new Frame(new Header(Version.REQUEST, this.flags, this.streamId, Opcode.OPTIONS), null);
    }

}
